### PR TITLE
rpmspectool: migrate to `python@3.12`

### PR DESCRIPTION
**Pull request by branchv**
rpmspectool: migrate to `python@3.12`

### CODE REVIEW

File: Formula/r/rpmspectool.rb [hunk 1]
```diff
@@ -15,7 +15,8 @@ class Rpmspectool < Formula
   depends_on :linux
   depends_on "python-argcomplete"
   depends_on "python-pycurl"
-  depends_on "python@3.11"
+  depends_on "python-setuptools"
```
Comment (carlocab) [inline Formula/r/rpmspectool.rb:18]: Remind me why this is still needed given that `python@3.12` still has a `setuptools` resource?

Comment (branchvincent) [reply Formula/r/rpmspectool.rb:18]: it isn't actually shipped with `python@3.12` any longer, we still have the resource but only to build `pip` from source in a temporary venv (same with `flit-core`)

https://github.com/Homebrew/homebrew-core/blob/7de456363e6e1d773cc065219fff8f5c24d4348c/Formula/p/python%403.12.rb#L279-L285

**Review by carlocab — APPROVED** [review]

File: Formula/r/rpmspectool.rb (hunk 1 above)
Comment (carlocab) [reply Formula/r/rpmspectool.rb:18]: In that case, maybe this is no longer necessary?

https://github.com/Homebrew/homebrew-core/blob/7de456363e6e1d773cc065219fff8f5c24d4348c/Formula/p/python%403.12.rb#L24-L26

CC @Bo98

Comment (branchvincent) [reply Formula/r/rpmspectool.rb:18]: yea shouldn't be

Comment (Bo98) [reply Formula/r/rpmspectool.rb:18]: Pretty sure it's a `sysconfigdata` thing and not setuptools specifically.

With that said, we should do better here as CLT requirement is only really necessary for things that interface with C compiling, which is not that much.

Possible idea is a relocation-like thing where we replace the SDK reference with a placeholder and write in the new SDK in `post_install`.

**Pull request merged**